web/components/templates/rateLimit/RateLimitRulesView.tsx

```diff
@@ -0,0 +1,380 @@
+import { P } from "@/components/ui/typography";
+import { Col } from "../../layout/common/col";
+import { useState } from "react";
+import RateLimitRuleModal from "./RateLimitRuleModal";
+import { useQuery, useMutation, useQueryClient } from "@tanstack/react-query";
+import { useOrg } from "@/components/layout/org/organizationContext";
+import { $JAWN_API } from "@/lib/clients/jawn";
+import { components } from "@/lib/clients/jawnTypes/private";
+import useNotification from "@/components/shared/notification/useNotification";
+import {
+  Table,
+  TableHeader,
+  TableRow,
+  TableHead,
+  TableBody,
+  TableCell,
+} from "@/components/ui/table";
+import { Button } from "@/components/ui/button";
+import { PiPlusBold } from "react-icons/pi";
+import { Badge } from "@/components/ui/badge";
+import { Trash2, Pencil } from "lucide-react";
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  AlertDialogTrigger,
+} from "@/components/ui/alert-dialog";
+import LoadingAnimation from "@/components/shared/loadingAnimation";
+
+type RateLimitRuleView = components["schemas"]["RateLimitRuleView"];
+type UpdateRateLimitRuleParams =
+  components["schemas"]["UpdateRateLimitRuleParams"];
+
+const RateLimitRulesView = () => {
+  const org = useOrg();
+  const [isCreateModalOpen, setIsCreateModalOpen] = useState(false);
+  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
+  const [editingRule, setEditingRule] = useState<RateLimitRuleView | null>(
+    null
+  );
+  const { setNotification } = useNotification();
+  const queryClient = useQueryClient();
+
+  const { data, isLoading, error, isError } = useQuery({
+    queryKey: ["rateLimits", org?.currentOrg?.id],
+    queryFn: async () => {
+      if (!org?.currentOrg?.id) {
+        return null; // Or throw an error, depending on desired behavior
+      }
+      const response = await $JAWN_API.GET("/v1/rate-limits");
+      return response;
+    },
+    enabled: !!org?.currentOrg?.id,
+  });
+
+  const deleteRuleMutation = useMutation({
+    mutationFn: async (ruleId: string) => {
+      const resp = await $JAWN_API.DELETE("/v1/rate-limits/{ruleId}", {
+        params: {
+          path: { ruleId },
+        },
+      });
+      if (resp.data?.error) {
+        throw new Error(resp.data.error);
+      }
+      return resp.data;
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: ["rateLimits", org?.currentOrg?.id],
+      });
+    },
+    // onError handled by try/catch in handleDelete
+  });
```

Comment:
same thing here, the only issue is you can't invalidate, so you will need to call refetch

```suggestion
  const deleteRuleMutation = $JAWN_API.useMutation("delete", "/v1/rate-limits/{ruleId}",{
        params: {
          path: { ruleId },
        },
    },
    onSuccess: () => {
       rateLimitRefetch()
    },
    // onError handled by try/catch in handleDelete
  });
```



I left a comment here a while ago: https://github.com/openapi-ts/openapi-typescript/issues/1806